app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,288 @@
+            }
+          });
+          return filtered;
+        };
+
+        var eventWatchCallback = function(eventData) {
+          var eventsByName = eventData.by('metadata.name');
+
+          notificationGroupsMap = formatFilteredEvents(eventsByName);
+          notificationGroups = sortNotificationGroups(notificationGroupsMap);
+
+          $rootScope.$apply(function () {
+            // NOTE: we are currently only showing one project in the drawer at a
+            // time. If we go back to multiple projects, we can eliminate the filter here
+            // and just pass the whole array as notificationGroups.
+            // if we do, we will have to handle group.open to keep track of what the
+            // user is viewing at the time & indicate to the user that the non-active
+            // project is "asleep"/not being watched.
+            drawer.notificationGroups = _.filter(notificationGroups, function(group) {
+              return group.project.metadata.name === $routeParams.project;
+            });
+          });
+        };
+
+        // TODO: atm these are not being added to the notificationGroups
+        var notificationWatchCallback = function(event, notification) {
+          if(!notification.lastTimestamp) {
+            notification.lastTimestamp = new Date();
+          }
+          eventsFromNotifications.push(notification);
+        };
+
+        var deregisterRootScopeWatches = function() {
+          _.each(rootScopeWatches, function(deregister) {
+            deregister();
+          });
+        };
+
+        var toggleDrawerVisibility = function(projectName) {
+          if(!projectName) {
+            drawer.drawerHidden = true;
+          }
+        };
+
+        var statusClasses = {
+          info: 'pficon pficon-info',
+          error: 'pficon pficon-error-circle-o',
+          warning: 'pficon pficon-warning-triangle-o',
+          ok: 'pficon pficon-ok'
+        };
+
+        var statuses = {
+          Normal: statusClasses.info,
+          Warning: statusClasses.warning
+        };
+
+        angular.extend(drawer, {
+          drawerHidden: true,
+          allowExpand: true,
+          drawerExpanded: false,
+          drawerTitle: 'Notifications',
+          counts: {},
+          onClose: function() {
+            drawer.drawerHidden = true;
+          },
+          notificationGroups: notificationGroups,
+          headingInclude: 'views/directives/notifications/heading.html',
+          subheadingInclude: 'views/directives/notifications/subheading.html',
+          notificationBodyInclude: 'views/directives/notifications/notification-body.html',
+          notificationFooterInclude: 'views/directives/notifications/notification-footer.html',
+          //
+          customScope: {
+            clearAll: function(group) {
+              _.each(group.notifications, function(notification) {
+                notification.unread = false;
+                EventsService.markRead(notification);
```

Comment:
Clear all is marking the notifications read. This should use 'Mark All Read' rather than "Clear All'